### PR TITLE
use ark_scale

### PR DESCRIPTION
**Pull request by achimcc**
Use [ark-scale](https://github.com/w3f/ark-scale) by @burdges for (de-)serialization.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2701753

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2701736

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2701737

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2702476

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2702475

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2702883

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2703169

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2703170

File: primitives/arkworks/src/utils.rs [hunk 1]
```diff
@@ -1,124 +1,150 @@
...
 
-/// Compute a projective scalar multiplication for twisted_edwards through arkworks
-pub fn mul_projective_te_generic<Group: TECurveConfig>(base: Vec<u8>, scalar: Vec<u8>) -> Vec<u8> {
-	let base = deserialize_argument::<twisted_edwards::Projective<Group>>(&base);
-	let scalar = deserialize_argument::<Vec<u64>>(&scalar);
+pub(crate) fn mul_projective_te_generic<Group: TECurveConfig>(
+	base: Vec<u8>,
+	scalar: Vec<u8>,
+) -> Result<Vec<u8>, ()> {
+	let base = <ArkScale<twisted_edwards::Projective<Group>> as Decode>::decode(
```
Comment (burdges) [inline primitives/arkworks/src/utils.rs:106]: `ark_scale::hazmat::ArkScaleProjective<twisted_edwards::Projective<..>>`

File: primitives/arkworks/src/utils.rs [hunk 2]
```diff
@@ -1,124 +1,150 @@
...
+pub(crate) fn mul_affine_generic<Group: SWCurveConfig>(
+	base: Vec<u8>,
+	scalar: Vec<u8>,
+) -> Result<Vec<u8>, ()> {
+	let base = <ArkScale<short_weierstrass::Affine<Group>> as Decode>::decode(
+		&mut base.clone().as_slice(),
+	)
+	.map_err(|_| ())?;
+	let scalar =
+		<ArkScale<Vec<u64>> as Decode>::decode(&mut scalar.clone().as_slice()).map_err(|_| ())?;
```
Comment (burdges) [inline primitives/arkworks/src/utils.rs:128]: We'll remove this and make `mul_affine` just call `mul_projective` in ark-substrate, but I guess that's a later PR.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2706764

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2706748

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2706763

**Pull request merged**